python/common/org/python/types/Object.java

```diff
@@ -979,4 +990,110 @@ public void __delitem__(org.python.Object index) {
     public org.python.Object __round__(org.python.Object ndigits) {
         throw new org.python.exceptions.AttributeError(this, "__round__");
     }
+
+    // Need to implement Is __eq__ Is Not __ne__
+    // In __contains__ (reversed operands) Not In __not_contains__ (reversed operands)
+    //
+    // for Is and Is Not, falls through to here only when either side is a constant in python
+    // which ends up as org.python.types.[Int|Float|Complex], otherwise it's dealt with
+    // by object reference comparison IF_ACMEQ. so we fall through to __eq__!
+    public enum CMP_OP {
+        GE(">=", "__ge__", "__le__"),
```

Comment:
It's rather strange mathematically for `ge/gt/le/lt`, but:

From [python/cpython/Objects/object.c at v3.6.0](https://github.com/python/cpython/blob/v3.6.0/Objects/object.c#L647):
```
/* Map rich comparison operators to their swapped version, e.g. LT <--> GT */
int _Py_SwappedOp[] = {Py_GT, Py_GE, Py_EQ, Py_NE, Py_LT, Py_LE};
static const char * const opstrings[] = {"<", "<=", "==", "!=", ">", ">="};
```

From [3. Data model – Python 3.5](https://docs.python.org/3.5/reference/datamodel.html?highlight=data%20model#object.__lt__):
> There are no swapped-argument versions of these methods (to be used when the left argument does not support the operation but the right argument does); rather, `__lt__()` and `__gt__()` are each other’s reflection, `__le__()` and `__ge__()` are each other’s reflection, and `__eq__()` and `__ne__()` are their own reflection.

As for `eq/ne`, let's say we're evaluating `x == y`. The reflection is in the sense that if `x.__eq__(y)` returns NotImplemented, then they will perform the reflected operation on y with x as parameter, i.e. `y.__eq__(x)`.